lib/megaphone_client.rb

```diff
@@ -1,11 +1,112 @@
-require "megaphone"
+require "megaphone_client/podcast"
+require "megaphone_client/episode"
 
+# @author Jay Arella
 module MegaphoneClient
-  class << self
-    attr_accessor :token, :network_id, :organization_id
+  class ConnectionError < StandardError
   end
-  
-  def self.setup
-    yield self
+
+  class << self
+    attr_accessor :api_base_url, :network_id, :organization_id, :token
+
+    # @option options [String] :api_base_url The api base url includes the domain and `/api` endpoint
+    # @option options [String] :network_id The id of the Megaphone network
+    # @option options [String] :organization_id The id of the Megaphone organization
+    # @option options [String] :token The api token taken from Megaphone's developer settings (https://developers.megaphone.fm/)
+    # @return a new instance of MegaphoneClient
+    # @example Initialize MegaphoneClient into an instance
+    #   @megaphone = MegaphoneClient.new({
+    #     network_id: '1234',
+    #     organization_id: '5678',
+    #     token: '910'
+    #   })
+    #
+    #   @megaphone #=> new MegaphoneClient
+
```

Comment:
Stylistic opinion, but I usually don't include a newline between inline comments the the function definition, so it's easier when reading the code to see what block of code the comments are attached to.

Reply:
This makes sense, will also add